api/pb/android/threads/src/main/java/io/textile/threads/TextileConfig.java

```diff
@@ -0,0 +1,114 @@
+import java.util.function.Consumer;
+
+public class TextileConfig implements Config {
+    public static String scheme = "https";
+    public static String host = "api.textile.io";
+    public static int port = 6446;
+    public static String session;
+    private static String token;
+    private static String deviceId;
+    public static ManagedChannel channel;
+    public String cloud = "https://cloud.textile.io:443/register";
+
+    /**
+     * TextileConfig provide auth wrappers to use hosted Thread services from Textile
+     * @param token your Project Token from Textile CLI
+     * @param deviceId a uuid for this app install.
+     */
+    public TextileConfig(String token, String deviceId) {
+        this.deviceId = deviceId;
+        this.token = token;
+    }
+    /**
+     * @return the current session information
+     */
+    public String getSession(){
+        return session;
+    }
+
+    /**
+     *
+     * @param session an existing session from getSession
+     */
+    public void setSession(String session){
+        this.session = session;
+    }
+    /**
+     * @return the gRPC managed channel
+     */
+    public ManagedChannel getChannel() {
+        return channel;
+    }
+    /**
+     * A private method to be used by the Client to initialize when instructed
+     * @param ready
+     */
+    public void init(Consumer<Boolean> ready) {
+        if (scheme == "http") {
+            channel = ManagedChannelBuilder
+                    .forAddress(host, port)
+                    .usePlaintext()
+                    .build();
+        } else {
+            channel = ManagedChannelBuilder
+                    .forAddress(host, port)
+                    .useTransportSecurity()
+                    .build();
+        }
+        if (this.session != null) {
+            // skip the session refresh if it's not null
+            ready.accept(true);
+            return;
+        }
+        this.refreshSession((result) -> {
+            this.session = result.session_id;
+            ready.accept(true);
+        });
+    }
+    public void refreshSession(Consumer<Session> callback) {
+        OkHttpClient client = new OkHttpClient();
+        client.setProtocols(Arrays.asList(Protocol.HTTP_1_1));
+        JsonObject json = new JsonObject();
+        json.addProperty("token", token);
+        json.addProperty("device_id",deviceId);
+        String jsonString = json.toString();
+        MediaType JSON = MediaType.parse("application/json; charset=utf-8");
+        RequestBody body = RequestBody.create(JSON, jsonString);
+        Request request = new Request.Builder()
+                .url(cloud)
+                .post(body)
+                .build();
+        client.newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(Request request, IOException throwable) {
+                throwable.printStackTrace();
+                System.out.println("Info: err " + throwable.getMessage());
```

Comment:
Yea, this is a bit of the problem with trying to use `Consumer`... There's no way to return the error from this async http back to the caller.

Reply:
okay, updated to use a Future